### PR TITLE
Postgres fixes, Harmonization improvements, parser fixes

### PR DESCRIPTION
**Pull request by sebix**
During the postgres fixes, the harmonization discussion came up:
The docs on harmonization (fields list) can now be automatically generated, along with types descriptions
all changes in harmonization have been applied to the bots
additional harmonization types: integer, float and Boolean
time.observation is added in collectors, as it is defined
Tests for some harmonization types

Various bot fixes and as always some more steps towards Python 3.

### CODE REVIEW

File: docs/Data-Harmonization.md [hunk 1]
```diff
@@ -306,10 +204,10 @@ Below, we have enumerated the minimum requirements for an actionable abuse event
...
 |Time|source time|Must|
 |Time|observation time|Must|
-|Identity|source_ip|Must*|
-|Identity|source_domain_name|Must*|
-|Identity|source_url|Must*|
-|Identity|source_email_address|Must*|
+|Identity|source.ip|Must*|
+|Identity|source.fqdn|Must*|
+|Identity|source.url|Must*|
+|Identity|source.email_address|Must*|
```
Comment (SYNchroACK) [inline docs/Data-Harmonization.md]: rename to source.account

File: intelmq/conf/harmonization.conf [hunk 2]
```diff
@@ -1,226 +1,357 @@
...
+            "description": "A malware family name in lower case.",
+            "regex": "[a-z ]+",
+            "type": "MalwareName"
+        },
+        "malware.version": {
+            "ascii": true,
+            "description": "A version string for an identified artifact generation, e.g. a crime-ware kit.",
+            "type": "String"
+        },
+        "misp_ip": {
```
Comment (SYNchroACK) [inline intelmq/conf/harmonization.conf]: so, we will not use 'extra' field ?

btw, change to misp_uuid as mentioned in #280

File: intelmq/conf/harmonization.conf [hunk 3]
```diff
@@ -1,226 +1,357 @@
...
+            "description": "The original line of the event from encoded in base64.",
+            "length": 100000,
+            "type": "Base64"
+        },
+        "rtir_id": {
+            "ascii": true,
+            "description": "Request Tracker Incident Response incident id.",
+            "type": "Integer"
+        },
+        "screenshot_url": {
```
Comment (SYNchroACK) [inline intelmq/conf/harmonization.conf]: I would remove that...

File: intelmq/conf/harmonization.conf [hunk 4]
```diff
@@ -1,226 +1,357 @@
...
+            "description": "Time reported by a source. Some sources only report a date, which may be used here if there is no better observation.",
+            "type": "DateTime"
+        },
+        "user_agent": {
+            "description": "Some feeds report the user agent string used by the host to access a malicious resource, such as a command and control server.",
+            "type": "String"
+        }
+    },
+    "report": {
+        "feed.code": {
```
Comment (SYNchroACK) [inline intelmq/conf/harmonization.conf]: I think is not needed anymore

Comment (sebix) [inline intelmq/conf/harmonization.conf]: @aaronkaplan what do you say?

Comment (SYNchroACK) [inline intelmq/conf/harmonization.conf]: @sebix forget that... lets merge....

Comment (aaronkaplan) [inline intelmq/conf/harmonization.conf]: ---
Mobile


> On 28.08.2015, at 16:34, Tomás Lima <notifications@github.com> wrote:
> 
> In intelmq/conf/harmonization.conf:
> 
> > +        },
> > +        "time.observation": {
> > +            "description": "The time a source bot saw the event. This timestamp becomes especially important should you perform your own attribution on a host DNS name for example. The mechanism to denote the attributed elements with reference to the source provided is detailed below in Reported Identity IOC.(ISO8660).",
> > +            "type": "DateTime"
> > +        },
> > +        "time.source": {
> > +            "description": "Time reported by a source. Some sources only report a date, which may be used here if there is no better observation.",
> > +            "type": "DateTime"
> > +        },
> > +        "user_agent": {
> > +            "description": "Some feeds report the user agent string used by the host to access a malicious resource, such as a command and control server.",
> > +            "type": "String"
> > +        }
> > +    },
> > +    "report": {
> > +        "feed.code": {
> I think is not needed anymore
> 

Other CERTs use a Feed Code to anonymize the feed reference. I'd keep it. It is optional anyway.

**Pull request merged**